protocol/x/vault/keeper/orders_test.go

```diff
@@ -287,34 +284,69 @@ func TestRefreshVaultClobOrders(t *testing.T) {
 			if tc.expectedErr != nil {
-				// Check that the error is as expected.
-				require.ErrorContains(t, err, tc.expectedErr.Error())
-				// Check that there's no stateful orders.
-				require.Len(t, allStatefulOrders, 0)
+				// Verify that no order is placed and chain doesn't halt.
+				require.Empty(t, tApp.App.ClobKeeper.GetAllStatefulOrders(ctx))
+				tApp.AdvanceToBlock(uint32(tApp.GetBlockHeight())+1, testapp.AdvanceToBlockOptions{})
 				return
-			} else {
-				// Check that there's no error.
-				require.NoError(t, err)
-				// Check that the number of orders is as expected.
-				params := tApp.App.VaultKeeper.GetParams(ctx)
-				require.Len(t, allStatefulOrders, int(params.Layers*2))
-				// Check that the orders are as expected.
-				expectedOrders, err := tApp.App.VaultKeeper.GetVaultClobOrders(ctx, tc.vaultId)
-				require.NoError(t, err)
-				for i := uint32(0); i < params.Layers*2; i++ {
-					require.Equal(t, *expectedOrders[i], allStatefulOrders[i])
-				}
 			}
+
+			// Vault should place its initial orders.
+			initialOrders := tApp.App.ClobKeeper.GetAllStatefulOrders(ctx)
+			require.Len(t, initialOrders, int(params.Layers*2))
+
+			// Advance to next block with no price updates / order matches and vault should not refresh its orders.
+			msgSender.Clear()
+			ctx = tApp.AdvanceToBlock(uint32(tApp.GetBlockHeight())+1, testapp.AdvanceToBlockOptions{})
+			require.Equal(
+				t,
+				initialOrders,
+				tApp.App.ClobKeeper.GetAllStatefulOrders(ctx),
+			)
+
+			// Advance to next block with price updates and vault should replace its old orders with new ones.
+			msgSender.Clear()
+			marketPrice, err := tApp.App.PricesKeeper.GetMarketPrice(ctx, tc.vaultId.Number)
+			require.NoError(t, err)
+			msgUpdateMarketPrices := &pricestypes.MsgUpdateMarketPrices{
+				MarketPriceUpdates: []*pricestypes.MsgUpdateMarketPrices_MarketPrice{
+					{
+						MarketId: tc.vaultId.Number,
+						Price:    marketPrice.Price * 2,
+					},
+				},
+			}
+			ctx = tApp.AdvanceToBlock(
+				uint32(tApp.GetBlockHeight())+1,
+				testapp.AdvanceToBlockOptions{
+					DeliverTxsOverride: [][]byte{
+						testtx.MustGetTxBytes(msgUpdateMarketPrices),
+					},
+				},
+			)
+			newOrders := tApp.App.ClobKeeper.GetAllStatefulOrders(ctx)
+			require.Len(t, newOrders, int(params.Layers*2))
+			for i, newOrder := range newOrders {
+				require.Equal(
+					t,
+					*tc.vaultId.GetClobOrderId(initialOrders[i].OrderId.ClientId ^ 1),
+					newOrder.OrderId,
+				)
+			}
+
+			// Advance to next block where vault orders have expired and vaults should place new orders.
+			ctx = tApp.AdvanceToBlock(
+				uint32(tApp.GetBlockHeight())+1,
+				testapp.AdvanceToBlockOptions{
+					BlockTime: ctx.BlockTime().Add(
+						time.Second * time.Duration(params.OrderExpirationSeconds+1),
+					),
+				},
+			)
+			newOrders = tApp.App.ClobKeeper.GetAllStatefulOrders(ctx)
+			require.Len(t, newOrders, int(params.Layers*2))
```

Comment:
Check that client IDs are different from before?

Reply:
Or check that the expiration timestamps on the orders are updated.